pip/src/interop.rs

```diff
@@ -55,12 +55,20 @@ impl<T> SourceResolver for ImportResolver<T>
 where
     T: FileSystem,
 {
-    fn resolve<P>(&self, path: P) -> miette::Result<(PathBuf, String)>
+    fn ctx(&mut self) -> &mut SourceResolverContext {
+        &mut self.ctx
+    }
+
+    fn resolve<P>(&mut self, path: P) -> miette::Result<(PathBuf, String), qsc::qasm::io::Error>
     where
         P: AsRef<Path>,
     {
         let path = self.path.join(path);
```

Comment:
Please use `self.fs.join` to avoid dumb issues path separators, and to resolve away `.` and `..`s in paths.

Reply:
https://github.com/microsoft/qsharp/pull/2303